### PR TITLE
use bluebird instead of Q for promises

### PR DESCRIPTION
**Pull request by dbushong**
* use bluebird instead of q
* fix tests for bluebird
* upgrade version requirements
* remove no-longer-necessary mentions of mocha-as-promised
* recompile JS with new coffee
* add dev notes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,5 +75,11 @@ it 'runs and fails asynchronously', ->
 (this may be bad style, depending on who you ask, but I find it useful if you
 have `it()`s with a lot of setup overhead)
 
+## Development
+
+* `src/aap.coffee` is the main library; it compiles to `lib/aap.js`.
+* The internal test suite uses a known-working copy: `lib/stable.js`.  When all your tests have passed successfully and you're ready to ship, go ahead and `cp lib/aap.js lib/stable.js` to have it available for internal testing next time around.
```
Comment (jkrems) [inline README.md]: Would adding itself as a dev dependency work? E.g. `"devDependencies": { "assertive-as-promised": "*" }`?

File: package.json [hunk 2]
```diff
@@ -12,14 +12,13 @@
     "test": "npm install && ./node_modules/.bin/mocha -R spec --compilers coffee:coffee-script/register test"
   },
   "author": "David Bushong <david+npm@bushong.net>",
-  "license": "BSD",
+  "license": "SEE LICENSE IN LICENSE",
```
Comment (jkrems) [inline package.json]: I always hate libraries that make me guess what the license actually is. I guess this is `BSD-3-Clause` in terms of SPDX identifiers?

**Comment by jkrems** [comment]
Two meta comments, the diff itself LGTM. Also, :fireworks: for `bluebird` :)

**Comment by jkrems** [comment]
:shipit:

**Comment by khoomeister** [comment]
Should https://github.com/groupon/assertive-as-promised/blob/e85b921f1be489b0330bef7e435b64ec529955e7/README.md be updated to reflect the change to bluebird?

**Comment by dbushong** [comment]
Ah, yes, probably.  What's the right-way to phrase it?  Should I actually say Bluebird-compatible, or something more high-minded, like "ES6-compliant promises"?

**Comment by khoomeister** [comment]
"ES6-compliant promises (through Bluebird)"?

**Comment by jkrems** [comment]
"standard-compliant promises", it's not ES6 anymore. ;)

**Comment by jkrems** [comment]
Also, github rightfully complaints about missing CI. Maybe throw in a `.travis.yml` in that README PR?

**Pull request merged**